harness/assert.js

```diff
@@ -79,3 +79,12 @@ assert.throws = function (expectedErrorConstructor, func, message) {
   message += 'Expected a ' + expectedErrorConstructor.name + ' to be thrown but no exception was thrown at all';
   $ERROR(message);
 };
+
+assert.throws.early = function(err, code) {
+  var wrappedCode = `function wrapperFn() { ${code} }`;
+  var ieval = eval;
+
+  assert.throws(err, () => { Function(wrappedCode); }, `Function: ${code}`);
+  assert.throws(err, () => { eval(wrappedCode); }, `eval: ${code}`);
+  assert.throws(err, () => { ieval(wrappedCode); }, `indirect eval: ${code}`);
+};
```

Comment:
In V8, this assert.throws.early will never get at preparser behavior. The Function constructor currently always invokes the parser, rather than the preparser, with the heuristic reasoning being that calls to the Function constructor are less likely to be dead code. If you want to make sure there's as good coverage for this as there is for the frontmatter early error declarations, you'll need host integration.

Do you have any particular reason to believe that invoking these three calls is more likely to get at an early error than doing what harnesses do for early errors? In V8, I'll be surprised if find much of a difference among them, beyond the semantic differences (global vs in a function to itself vs directly in assert.throws.early).